### ISSUE TITLE
Upgrade Mrcms

### ISSUE THREAD

**Issue by Sourcenew** [MrCMS/MrCMS#91]
do you have any plan to upgrade Mrcms to .net-core or switch to entity-framework in future?

**Comment by Will-Bill** [comment]
Nearly there - sneak peak:

https://i.imgur.com/NGXwCBt.png

If anyone would like to help with beta testing / functionality please let me know here.

**Comment by Sourcenew** [comment]
i can suggest some features for multilingual and multi calendar support.
and for testing. i can help you. it's good news dear.

and if Mrcms go's to use entity-framework, i can write migrate application

how i can help you?

**Comment by Sourcenew** [comment]
one more thing, i wrote my ACL,

upgraded version of your acl, but i don't use static list of acl, i collect acl's from action result attributes.

and menu was rendred automatically from you role.

sorry my english not very good

<img src="https://i.imgur.com/tmBNbwY.png" width="100%" />

<img src="https://i.imgur.com/3pOYaP2.png" width="100%" />

<img src="https://i.imgur.com/eURiBaB.png" width="100%" />

can you think we can use it in mrcms? or ...

**Comment by Will-Bill** [comment]
We have done something similar in the new system. Also builds up a breadcrumb. Will try set up a private beta next week.

Staying with nHibernate for ORM for now.

**Comment by codeendevor** [comment]
@Will-Bill I would like to be a part of this, I have a complete Admin and Front end theme ready for MrCMS but was waiting for the Core project as you had told me this is what we had been moving to.

**Comment by Sourcenew** [comment]
most of my problems of current Mrcms was NHibernate, i hope this be change in future.

**Comment by Sourcenew** [comment]
it's my problem not Mrcms, i have to learn how use NHibernate

**Comment by codeendevor** [comment]
@Will-Bill, is it possible to get any Images you have for MrCMS? Need for the new themes. Once complete i will publish for your review.

**Comment by codeendevor** [comment]
> Nearly there - sneak peak:
> 
> https://i.imgur.com/NGXwCBt.png
> 
> If anyone would like to help with beta testing / functionality please let me know here.

Im In. Just let me know when.

**Comment by Will-Bill** [comment]
@codeendevor do you mean logos for Mr CMS?

Quick update - pushed latest DNC version of Mr CMS here: https://github.com/MrCMS/MrCMS/tree/feature/dotnetcore

Admin new look: https://i.imgur.com/axWnWt7.png 

Couple notes:

- There is still work to do on this, I would only use it for experimental purposes for now
- Apps are now in their own libraries - views are embedded/compiled in their libraries making them portable. 
- Admin is now and 'App' also 
- Themes are also their own libraries, again making them portable
- At the moment there are admin views in the MrCMS.Web/Areas/Admin location just for ease of migration to bootstrap 4. Once complete the views will be moved back to Admin App.
- Views will be looked up in MrCMS.Web, then Theme, Then App locations
- Admin is now Bootstrap 4 using Admin LTE 
- Web Push Notifications have been integrated. You can play with this by clicking subscribe on the homepage, then allow in your browser. You can then send a push notification from the admin nav bar top right. This will be useful for blogs/news sites and also eCommerce for sending out latest deals and top news etc
- We have partially tested on Linux and Mac and seems to work OK. Our plan is to make this run on Linux perfectly so Azure hosting of App Services is much cheaper
- Avatars can be uploaded to personalise user experience
- Forms have been separated into their own section in admin. You now can embed the same form in multiple locations using [form id="{id}"] swapping {id} for the id of the form

Let me know your thoughts, and if you come across issues feel free to fix and do a pull request :)

**Comment by codeendevor** [comment]
yes and I was hoping for some of the images from the MrCMS site so i can add them to the project build and install project

**Comment by codeendevor** [comment]
How did you run this? was it through IIS? Having issues getting past Setup screen, Are all the SQL servers available for use?

**Comment by johancruyff2019** [comment]
i can't run it, it goes to error 500 (no syntax error)

question: for per App we need creat new project?

**Comment by codeendevor** [comment]
did you try and run from command line, I did, was able to go through setup then project stops, It did create the database. in trying to go through everything now to see if i can find whats up. I do know i had to change the path for System.Private.XML to my version of Core 2

**Comment by codeendevor** [comment]
I did fix all the warnings so we are compliant with net core 2.1 and up. but do not want to commit till i have this running. ill post my findings.

**Comment by Will-Bill** [comment]
@Sourcenew 

"question: for per App we need creat new project?"

Its just a class library as so: https://i.imgur.com/TRAGX1Q.png

Here is an example csproj:

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netcoreapp2.2</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <EmbeddedResource Include="Areas\Admin\Views\**" />
    <EmbeddedResource Include="Views\**" />
    <EmbeddedResource Include="wwwroot\**" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="X.PagedList.Mvc.Core" Version="7.6.0" />
  </ItemGroup>

  <ItemGroup>
    <ProjectReference Include="..\..\Lib\MrCMS.Web.Apps.Admin.Infrastructure\MrCMS.Web.Apps.Admin.Infrastructure.csproj" />
    <ProjectReference Include="..\..\Lib\MrCMS\MrCMS.csproj" />
  </ItemGroup>

  <ItemGroup>
    <Folder Include="wwwroot\scripts\" />
  </ItemGroup>

</Project>
```

In terms of you not being able to run the project - have you got the latest SDK installed (SDK v2.2.103)?

**Comment by codeendevor** [comment]
Yes we are on the latest 2.2.1

**Comment by johancruyff2019** [comment]
Yes and i have latest updates of VS2017

**Comment by Will-Bill** [comment]
Download the SDK v2.2.**103** version of SDK from https://dotnet.microsoft.com/download - i was getting some errors with 102 version.

I have just tried building from VS 2017/Rider IDE and its fine, both from a fresh checkout.  Also do a pull on the latest changes there.

Finally make sure you are running src/MrCMS.Web project

**Comment by codeendevor** [comment]
will try, i do have SDK v2.2.103 already.

**Comment by Will-Bill** [comment]
OK. So what errors are you getting?

You did say "did you try and run from command line, I did, was able to go through setup then **project stops**"

Sadly this is because the dotnet process quits after installation and doesn't automatically re-run. Simply run dotnet run again and the website will be up and running.

**Comment by johancruyff2019** [comment]
Should I manually create databases and tables?

**Comment by johancruyff2019** [comment]
and can you connect to my pc through teamviewer or anydesk?

**Comment by codeendevor** [comment]
i did notice the install process went very fast the first few times i did the install, I did have to point System.Private.XML to my version of Core 2.2.103 path, Should be an easier way to reference core files. after this new pull tonight time the installer seamed to take longer but still no luck. 
`'dotnet.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\2.2.1\System.Data.SqlClient.dll'. Cannot find or open the PDB file.
'dotnet.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\2.2.1\System.Security.Principal.Windows.dll'. Cannot find or open the PDB file.
'dotnet.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\2.2.1\System.Text.Encoding.CodePages.dll'. Cannot find or open the PDB file.
'dotnet.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\2.2.1\System.Data.dll'. Cannot find or open the PDB file.
'dotnet.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\2.2.1\System.Linq.Queryable.dll'. Cannot find or open the PDB file.
'dotnet.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\2.2.1\System.Reflection.Extensions.dll'. Cannot find or open the PDB file.
'dotnet.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\2.2.1\System.Runtime.Serialization.Primitives.dll'. Cannot find or open the PDB file.
'dotnet.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\2.2.1\Microsoft.AspNetCore.Http.Extensions.dll'. Cannot find or open the PDB file.
MrCMS.Services.Resources.StringResourceProvider:Information: Resource list empty
Failed to load resource: the server responded with a status of 404 () [https://localhost:44301/]
Microsoft.AspNetCore.Hosting.Internal.WebHost:Information: Request finished in 18429.8979ms 404 
The thread 0x5e48 has exited with code 0 (0x0).
The thread 0x51c has exited with code 0 (0x0).
The thread 0x3d90 has exited with code 0 (0x0).`

I need to go through the project file and change locations of all the core files.

**Comment by codeendevor** [comment]
```
MrCMS.Services.Resources.StringResourceProvider:Information: Resource list empty
Failed to load resource: the server responded with a status of 404 () [https://localhost:44301/]
Microsoft.AspNetCore.Hosting.Internal.WebHost:Information: Request finished in 11443.3644ms 404 
```

Are we still writing files to the appdata folder, if so maybe this is my issue, appdata folder was created but no files in it after setup.

**Comment by codeendevor** [comment]
Database was created and populated, and appsettings.Development.json file created pointing to database.

**Comment by johancruyff2019** [comment]
how? please help me

**Comment by codeendevor** [comment]
I did have to change to 127.0.0.1 for some reason i was getting permission errors on local host. but still no luck. `Resource list empty`
```
C:\Projects\GitHub\MrCMS_Core\src\MrCMS.Web>dotnet run
C:\Program Files\dotnet\sdk\2.2.103\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(153,5): warning NETSDK1071: A PackageReference to 'Microsoft.AspNetCore.App' specified a Version of `2.2.1`. Specifying the version of this package is not recommended. For more information, see https://aka.ms/sdkimplicitrefs [C:\Projects\GitHub\MrCMS_Core\src\MrCMS.Web\MrCMS.Web.csproj]
C:\Program Files\dotnet\sdk\2.2.103\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(153,5): warning NETSDK1071: A PackageReference to 'Microsoft.AspNetCore.App' specified a Version of `2.2.1`. Specifying the version of this package is not recommended. For more information, see https://aka.ms/sdkimplicitrefs [C:\Projects\GitHub\MrCMS_Core\src\Lib\MrCMS\MrCMS.csproj]
info: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[0]
      User profile is available. Using 'C:\Users\Rich\AppData\Local\ASP.NET\DataProtection-Keys' as key repository and Windows DPAPI to encrypt keys at rest.
Hosting environment: Development
Content root path: C:\Projects\GitHub\MrCMS_Core\src\MrCMS.Web
Now listening on: https://127.0.0.1:44301
Now listening on: http://127.0.0.1:44300
Application started. Press Ctrl+C to shut down.
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET http://127.0.0.1:44300/
info: MrCMS.Services.Resources.StringResourceProvider[0]
      Resource list empty
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 6444.5447ms 404
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET http://127.0.0.1:44300/
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 27.2084ms 404
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET http://127.0.0.1:44300/
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 41.1709ms 404
```

**Comment by Will-Bill** [comment]
try on https and make sure the dev cert is trusted.

@Sourcenew send me teamviewer details and I will try - can add me on skype if you want thought.will

**Comment by codeendevor** [comment]
Ok Cert is good: here is output from begining.
`C:\Projects\GitHub\MrCMS_Core\src\MrCMS.Web>dotnet run
C:\Program Files\dotnet\sdk\2.2.103\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(153,5): warning NETSDK1071: A PackageReference to 'Microsoft.AspNetCore.App' specified a Version of `2.2.1`. Specifying the version of this package is not recommended. For more information, see https://aka.ms/sdkimplicitrefs [C:\Projects\GitHub\MrCMS_Core\src\MrCMS.Web\MrCMS.Web.csproj]
C:\Program Files\dotnet\sdk\2.2.103\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(153,5): warning NETSDK1071: A PackageReference to 'Microsoft.AspNetCore.App' specified a Version of `2.2.1`. Specifying the version of this package is not recommended. For more information, see https://aka.ms/sdkimplicitrefs [C:\Projects\GitHub\MrCMS_Core\src\Lib\MrCMS\MrCMS.csproj]
info: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[0]
      User profile is available. Using 'C:\Users\Rich\AppData\Local\ASP.NET\DataProtection-Keys' as key repository and Windows DPAPI to encrypt keys at rest.
Hosting environment: Development
Content root path: C:\Projects\GitHub\MrCMS_Core\src\MrCMS.Web
Now listening on: https://localhost:44301
Now listening on: http://localhost:44300
Application started. Press Ctrl+C to shut down.
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET https://localhost:44301/
info: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[1]
      Route matched with {action = "Setup", controller = "Install", area = ""}. Executing action MrCMS.Installation.Controllers.InstallController.Setup (MrCMS)
info: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[1]
      Executing action method MrCMS.Installation.Controllers.InstallController.Setup (MrCMS) - Validation state: Valid
info: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[2]
      Executed action method MrCMS.Installation.Controllers.InstallController.Setup (MrCMS), returned result Microsoft.AspNetCore.Mvc.ViewResult in 136.9874ms.
info: Microsoft.AspNetCore.Mvc.ViewFeatures.ViewResultExecutor[1]
      Executing ViewResult, running view Setup.
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET https://localhost:44301/bootstrap/css/bootstrap.min.css
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET https://localhost:44301/Images/logo.png
info: Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware[2]
      Sending file. Request path: '/Images/logo.png'. Physical path: 'N/A'
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 10.23ms 200 image/png
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET https://localhost:44301/style.css
info: Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware[2]
      Sending file. Request path: '/style.css'. Physical path: 'N/A'
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 1.3957ms 200 text/css
info: Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware[2]
      Sending file. Request path: '/bootstrap/css/bootstrap.min.css'. Physical path: 'N/A'
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 95.6199ms 200 text/css
info: Microsoft.AspNetCore.Mvc.ViewFeatures.ViewResultExecutor[4]
      Executed ViewResult - view Setup executed in 3524.1671ms.
info: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[2]
      Executed action MrCMS.Installation.Controllers.InstallController.Setup (MrCMS) in 3725.3661ms
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 4078.2531ms 200 text/html; charset=utf-8
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET https://localhost:44301/images/install-synchronizing.gif
info: Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware[2]
      Sending file. Request path: '/images/install-synchronizing.gif'. Physical path: 'N/A'
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 0.4437ms 200 image/gif
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET https://localhost:44301/scripts/jquery-2.1.3.min.js
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET https://localhost:44301/scripts/jquery.validate.min.js
info: Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware[2]
      Sending file. Request path: '/scripts/jquery.validate.min.js'. Physical path: 'N/A'
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 0.8966ms 200 application/javascript
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET https://localhost:44301/scripts/jquery.validate.unobtrusive.min.js
info: Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware[2]
      Sending file. Request path: '/scripts/jquery.validate.unobtrusive.min.js'. Physical path: 'N/A'
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 0.3262ms 200 application/javascript
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET https://localhost:44301/scripts/install.js
info: Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware[2]
      Sending file. Request path: '/scripts/install.js'. Physical path: 'N/A'
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 0.2603ms 200 application/javascript
info: Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware[2]
      Sending file. Request path: '/scripts/jquery-2.1.3.min.js'. Physical path: 'N/A'
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 34.2244ms 200 application/javascript
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET https://localhost:44301/favicon.ico
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 4.5079ms 302
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET https://localhost:44301/
info: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[1]
      Route matched with {action = "Setup", controller = "Install", area = ""}. Executing action MrCMS.Installation.Controllers.InstallController.Setup (MrCMS)
info: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[1]
      Executing action method MrCMS.Installation.Controllers.InstallController.Setup (MrCMS) - Validation state: Valid
info: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[2]
      Executed action method MrCMS.Installation.Controllers.InstallController.Setup (MrCMS), returned result Microsoft.AspNetCore.Mvc.ViewResult in 0.9119ms.
info: Microsoft.AspNetCore.Mvc.ViewFeatures.ViewResultExecutor[1]
      Executing ViewResult, running view Setup.
info: Microsoft.AspNetCore.Mvc.ViewFeatures.ViewResultExecutor[4]
      Executed ViewResult - view Setup executed in 17.3088ms.
info: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[2]
      Executed action MrCMS.Installation.Controllers.InstallController.Setup (MrCMS) in 23.1494ms
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 26.5472ms 200 text/html; charset=utf-8
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 POST https://localhost:44301/ application/x-www-form-urlencoded 1024
info: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[1]
      Route matched with {action = "Setup", controller = "Install", area = ""}. Executing action MrCMS.Installation.Controllers.InstallController.Setup (MrCMS)
info: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[1]
      Executing action method MrCMS.Installation.Controllers.InstallController.Setup (MrCMS) with arguments (MrCMS.Installation.Models.InstallModel) - Validation state: Valid
info: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[2]
      Executed action method MrCMS.Installation.Controllers.InstallController.Setup (MrCMS), returned result Microsoft.AspNetCore.Mvc.ViewResult in 26300.5168ms.
info: Microsoft.AspNetCore.Mvc.ViewFeatures.ViewResultExecutor[1]
      Executing ViewResult, running view Success.
info: Microsoft.AspNetCore.Mvc.ViewFeatures.ViewResultExecutor[4]
      Executed ViewResult - view Success executed in 1235.2106ms.
Application is shutting down...

C:\Projects\GitHub\MrCMS_Core\src\MrCMS.Web>dotnet run
C:\Program Files\dotnet\sdk\2.2.103\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(153,5): warning NETSDK1071: A PackageReference to 'Microsoft.AspNetCore.App' specified a Version of `2.2.1`. Specifying the version of this package is not recommended. For more information, see https://aka.ms/sdkimplicitrefs [C:\Projects\GitHub\MrCMS_Core\src\MrCMS.Web\MrCMS.Web.csproj]
C:\Program Files\dotnet\sdk\2.2.103\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(153,5): warning NETSDK1071: A PackageReference to 'Microsoft.AspNetCore.App' specified a Version of `2.2.1`. Specifying the version of this package is not recommended. For more information, see https://aka.ms/sdkimplicitrefs [C:\Projects\GitHub\MrCMS_Core\src\Lib\MrCMS\MrCMS.csproj]
info: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[0]
      User profile is available. Using 'C:\Users\Rich\AppData\Local\ASP.NET\DataProtection-Keys' as key repository and Windows DPAPI to encrypt keys at rest.
Hosting environment: Development
Content root path: C:\Projects\GitHub\MrCMS_Core\src\MrCMS.Web
Now listening on: https://localhost:44301
Now listening on: http://localhost:44300
Application started. Press Ctrl+C to shut down.
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET https://localhost:44301/
info: MrCMS.Services.Resources.StringResourceProvider[0]
      Resource list empty
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 6339.6583ms 404`

**Comment by Will-Bill** [comment]
@codeendevor try again but use SQL Lite, see if that works

**Comment by Will-Bill** [comment]
What DatabaseProviderType is set in appsettings?

**Comment by codeendevor** [comment]
cert is good.
![image](https://user-images.githubusercontent.com/13207023/51977920-ba20a600-2456-11e9-9229-6824d2c7ea09.png)
Install Page:
![image](https://user-images.githubusercontent.com/13207023/51978027-f6540680-2456-11e9-82d0-723c5c2e6345.png)
Result:
![image](https://user-images.githubusercontent.com/13207023/51978279-c6f1c980-2457-11e9-97e0-e64263b3d2d0.png)

I did for a second get the success page before it shut down.

**Comment by codeendevor** [comment]
Microsoft.AspNetCore.Hosting.Internal.WebHost:Information: Request starting HTTP/1.1 DEBUG http://localhost:44301/  0
Microsoft.AspNetCore.Hosting.Internal.WebHost:Information: Request starting HTTP/1.1 GET http://localhost:44301/  
Microsoft.AspNetCore.Hosting.Internal.WebHost:Information: Request finished in 72.0714ms 200 
Exception thrown: 'NHibernate.SchemaValidationException' in NHibernate.dll
The thread 0x24c0 has exited with code 0 (0x0).

**Comment by Will-Bill** [comment]
It will shutdown after install. Needs to be ran again.

What db settings you using?

**Comment by codeendevor** [comment]
Shouldnt the request for the following be SSL https:
Microsoft.AspNetCore.Hosting.Internal.WebHost:Information: Request starting HTTP/1.1 GET http://localhost:44301/  
Microsoft.AspNetCore.Hosting.Internal.WebHost:Information: Request starting HTTP/1.1 DEBUG http://localhost:44301/  0

**Comment by codeendevor** [comment]
tried both sql server 2014 and the light you recommended, and yes the server was listed properly in the dev file

**Comment by codeendevor** [comment]
the can not be found page is after running again

**Comment by Will-Bill** [comment]
my work colleague had an issue until he restarted his machine. Maybe try that? I cannot reproduce.

What version of windows are you on?

**Comment by codeendevor** [comment]
Windows 10, Server 2012, Server 2016, tried both database types. all same result. I have built other core 2.2 projects to test and they work fine. Maybe im missing a step as i thing it is not working with core for some reason. Even though the site does not display if i go to the admin page i can see it send me to login page and same results. Resource list empty.

**Comment by Will-Bill** [comment]
I cannot reproduce this - have tried setting same timezone/culture. Windows, Linux, Mac -nothing.

The error message is coming from line 55 in StringResourceProvider - but I am not sure why resources have not been populated.

Do a fresh checkout of our branch and try once more (we've changed a few things since) - if still no success, look to see if any resources are in the database / see if inserts are being tried and failing.

**Comment by codeendevor** [comment]
Will try, And im sure im doing something wrong.

**Comment by codeendevor** [comment]
![image](https://user-images.githubusercontent.com/13207023/52393595-c6b78680-2a74-11e9-8f06-d7f91abc44a8.png)

**Comment by codeendevor** [comment]
Ok so here is what i have found. All the documents in the database had been unpublished, They had a publish date but published was all false. once i set them to true i now have access.

![image](https://user-images.githubusercontent.com/13207023/52397395-62e88a00-2a83-11e9-899d-cc875534d9ae.png)

**Comment by Will-Bill** [comment]
Strange - again, not something happening for us.

FYI the front end has been upgraded to Bootstrap 4.0 so you are a few commits behind.

**Comment by codeendevor** [comment]
humm i just did a sync. strange ill have to try again. same branch?

**Comment by codeendevor** [comment]
Yep got it. Looks great, And i like the content blocks. What can i do to help?

![image](https://user-images.githubusercontent.com/13207023/52398326-a1337880-2a86-11e9-9592-19c75130afa3.png)

**Comment by Will-Bill** [comment]
Ah cool!

The standard form rendering system needs the option Bootstrap 4 adding. In FormRenderingType the type Boostrap4 has been added, but the renderers need updating. I.E In MrCMS/Shortcodes/Forms there are a few classes which take care of rendering form items which are built using the form builder. See CheckBoxListRenderer.cs, DropDownListRenderer.cs etc - CheckBoxListRenderer is the only one I think we've updated so far - see code block:

`else if (formRenderingType == FormRenderingType.Bootstrap4)
                {
                    var checkboxContainer = new TagBuilder("div");
                    cbLabelBuilder.AddCssClass("form-check-label");
                    checkboxContainer.AddCssClass("form-check");
                    checkboxContainer.InnerHtml.AppendHtml(checkboxBuilder);
                    checkboxContainer.InnerHtml.AppendHtml(cbLabelBuilder);
                    tagBuilder.InnerHtml.AppendHtml(checkboxContainer);
                }`

Basically these form renderers need updating for the Bootstrap 4 forms: https://getbootstrap.com/docs/4.2/components/forms/

I don't think there are huge changes between 3 and 4 - mostly with radios and checkbox/checkbox lists.

But if you'd like to contribute - this would be really useful thing to help with.

**Comment by codeendevor** [comment]
Great, Ill get started tonight.

**Comment by carlospence** [comment]
Hello, Good evening

I am thinking of a feature in which Web API's can be automatically generated from Services (Auto API Controllers) and be display using Swagger for standalone front end consumption (SPA or Mobile)

**Comment by johancruyff2019** [comment]
I think it's possible,  i do something like that before now in my cms, we can return full document model and put our custom model in front side, get model, list (by filter and pagination)

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by carlospence** [comment]
I will work on it and send a PR when am done with Proof of Concept

**Comment by johancruyff2019** [comment]
👌❤️
Good luck
________________________________

**Comment by carlospence** [comment]
@Will-Bill At what point do you intend merging the two Admin Views into one because I think it is better it is done now.
